utils/oomparser/oomparser.go

```diff
@@ -104,15 +103,10 @@ func checkIfStartOfOomMessages(line string) (bool, error) {
 // lines are checked against a regexp to check for the pid, process name, etc.
 // At the end of an oom message group, AnalyzeLines adds the new oomInstance to
 // oomLog
-func (self *OomParser) analyzeLines(outPipe io.ReadCloser, outStream chan *OomInstance) {
-	ioreader := bufio.NewReader(outPipe)
+func (self *OomParser) analyzeLines(ioreader *bufio.Reader, outStream chan *OomInstance) {
 	line, err := ioreader.ReadString('\n')
 	for err == nil {
```

Comment:
This code is getting a bit hard to read, let's try to simplify it. We will never break out of this loop so lets make it an infinite loop with a read at the beginning. We should also only wait while the error is EOF:

```Go
var line string
var err error
for true {
    // Wait for more content to show up in the file.
    for line, err = ioreader.ReadString('\n'); err != nil && err == io.EOF; {
        time.Sleep(100 * time.Millisecond)
    }

    // Do work...
}
```